libtiff/libtiff_ctypes.py

```diff
@@ -786,59 +831,54 @@ def read_tiles(self, dtype=np.uint8):
         num_irows = self.GetField("ImageLength")
         if num_irows is None:
             num_irows = 1
-        num_idepth = self.GetField("ImageDepth")
-        if num_idepth is None:
-            num_idepth = 1
-
-        if num_idepth == 1 and num_irows == 1:
-            # 1D
-            full_image = np.zeros((num_icols,), dtype=dtype)
-        elif num_idepth == 1:
-            # 2D
-            full_image = np.zeros((num_irows, num_icols), dtype=dtype)
-        else:
-            # 3D
-            full_image = np.zeros((num_idepth, num_irows, num_icols),
-                                  dtype=dtype)
-
-        tmp_tile = np.zeros((num_trows, num_tcols), dtype=dtype)
-        tmp_tile = np.ascontiguousarray(tmp_tile)
-        for z in range(0, num_idepth):
-            for y in range(0, num_irows, num_trows):
-                for x in range(0, num_icols, num_tcols):
-                    r = self.ReadTile(tmp_tile.ctypes.data, x, y, z, 0)
+        # this number includes extra samples
+        samples_pp = self.GetField('SamplesPerPixel')
+        if samples_pp is None:  # default is 1
+            samples_pp = 1
+        planar_config = self.GetField('PlanarConfig')
+        if planar_config is None:  # default is contig
+            planar_config = PLANARCONFIG_CONTIG
+
```

Comment:
Ok, now I think I understood it. But I'm not so sure if it can work. Reading tiles in images with multiple samples per pixels, or reading tiles in images with multiple planes (PLANARCONFIG_SEPARATE) are done differently in libtiff. 
The definition of TIFFReadTile is:
`tsize_t TIFFReadTile(TIFF* tif, tdata_t buf, uint32 x, uint32 y, uint32 z, tsample_t sample)`
When we want a different image depth, we pass the image depth in the `z` parameter. When we want a different plane, when samples_pp > 1, we pass the plane index in the `sample` parameter.